tests/codegen/array-map.rs

```diff
@@ -21,7 +21,7 @@ pub fn short_integer_map(x: [u32; 8]) -> [u32; 8] {
 pub fn short_integer_zip_map(x: [u32; 8], y: [u32; 8]) -> [u32; 8] {
     // CHECK: %[[A:.+]] = load <8 x i32>
     // CHECK: %[[B:.+]] = load <8 x i32>
-    // CHECK: sub <8 x i32> %[[A]], %[[B]]
+    // CHECK: sub <8 x i32> %[[B]], %[[A]]
```

Comment:
uh, what happened here? Is this just a reordering of the loads but not a reordering of the actual subtraction?

Reply:
r=me after making sure that this test doesn't get miscompiled

Reply:
I don't know exactly what is going on here but the test isn't miscompiled. It's our aggressive abbreviations in this codegen test that make it look suspicious. Using -Zmir-opt-level=3 to simulate the effect of this PR because it lifts the inline attribute requirement: https://godbolt.org/z/EP7aTcz1x

Before:
```llvm
define void @_ZN7example21short_integer_zip_map17hbd5b1e22189a0746E(ptr noalias nocapture noundef writeonly sret([8 x i32]) dereferenceable(32) %0, ptr noalias nocapture noundef readonly dereferenceable(32) %x, ptr noalias nocapture noundef readonly dereferenceable(32) %y) unnamed_addr #0 personality ptr @rust_eh_personality !dbg !6 {
  %1 = load <8 x i32>, ptr %x, align 4, !dbg !11
  %2 = load <8 x i32>, ptr %y, align 4, !dbg !11
  %3 = sub <8 x i32> %1, %2, !dbg !12
  store <8 x i32> %3, ptr %0, align 4, !dbg !62, !alias.scope !63, !noalias !66
  ret void, !dbg !68
}
```
After:
```llvm
define void @_ZN7example21short_integer_zip_map17hbd5b1e22189a0746E(ptr noalias nocapture noundef writeonly sret([8 x i32]) dereferenceable(32) %0, ptr noalias nocapture noundef readonly dereferenceable(32) %x, ptr noalias nocapture noundef readonly dereferenceable(32) %y) unnamed_addr #0 personality ptr @rust_eh_personality !dbg !6 {
  %1 = load <8 x i32>, ptr %y, align 4, !dbg !11
  %2 = load <8 x i32>, ptr %x, align 4, !dbg !18
  %3 = sub <8 x i32> %2, %1, !dbg !19
  store <8 x i32> %3, ptr %0, align 4, !dbg !65, !alias.scope !66, !noalias !69
  ret void, !dbg !71
}
```